synapse/storage/databases/main/pusher.py

```diff
@@ -89,6 +89,19 @@ def _decode_pushers_rows(self, rows: Iterable[dict]) -> Iterator[PusherConfig]:
                 )
                 continue
 
+            # Pushers created while support for the 'enabled' field is not active
+            # (either because they were created before said support existed or because
+            # they were created while the experimental implementation is turned off)
+            # will have the 'enabled' column set to NULL, which needs to be interpreted
+            # as True.
+            if r["enabled"] is None:
+                r["enabled"] = True
```

Comment:
I thought we had a COALESCE somewhere for this? Don't mind it being here too, but it'd be nice if there was just one place responsible for worrying about NULLs

Reply:
This method is called by two methods: `get_enabled_pushers` and `get_pushers_by`. The former has a `COALESCE` (which was easy to add since it just calls `txn.execute` with some SQL) but the latter uses `simple_select_list` with a caller-defined `keyvalues`, and thus does not have a `COALESCE`. I've rewritten `get_pushers_by` to inline the query (because I thought giving `simple_select_list` `COALESCE(enabled, TRUE) AS enabled` as a column name was a bit hacky).